services/app-api/resolvers/indexSubmissions2.ts

```diff
@@ -0,0 +1,59 @@
+import { ApolloError, ForbiddenError } from 'apollo-server-lambda'
+import {
+    isStateUser,
+    Submission2Type
+} from '../../app-web/src/common-code/domain-models'
+import { QueryResolvers } from '../gen/gqlServer'
+import { logError, logSuccess } from '../logger'
+import { isStoreError, Store} from '../postgres'
+
+export function indexSubmissions2Resolver(
+    store: Store
+): QueryResolvers['indexSubmissions2'] {
+    return async (_parent, _args, context) => {
+        // This resolver is only callable by state users
+        if (!isStateUser(context.user)) {
+            logError(
+                'indexSubmissions',
+                'user not authorized to fetch state data'
+            )
+            throw new ForbiddenError('user not authorized to fetch state data')
+        }
+
+        // fetch from the store
+        const results = await store.findAllSubmissionsWithRevisions(context.user.state_code)
+
+        if (isStoreError(results)) {
+            if (results.code === 'WRONG_STATUS') {
```

Comment:
I think this isn’t relevant anymore. Submission2 postgres functions don’t specifiy draft/state submisssions.